R/add_rcsi.R

```diff
@@ -0,0 +1,90 @@
+#' Add indicator for reduced Household CSI Score(rcsi)
+#'
+#' @param data dataset
+#' @param rCSILessQlty Column representing question- During the last 7 days, were there days (and, if so, how many) when your household had to rely on less preferred and less expensive food to cope with a lack of food or money to buy it?
+#' @param rCSIBorrow   Column representing question- During the last 7 days, were there days (and, if so, how many) when your household had to borrow food or rely on help from a relative or friend to cope with a lack of food or money to buy it?
+#' @param rCSIMealSize Column representing question- During the last 7 days, were there days (and, if so, how many) when your household had to limit portion size of meals at meal times to cope with a lack of food or money to buy it?
+#' @param rCSIMealAdult Column representing question- During the last 7 days, were there days (and, if so, how many) when your household had to restrict consumption by adults in order for small children to eat to cope with a lack of food or money to buy it?
+#' @param rCSIMealNb Column representing question - During the last 7 days, were there days (and, if so, how many) when your household had to reduce number of meals eaten in a day to cope with a lack of food or money to buy it?
+#' @param new_colname The prefix for the new columns. It has to be a string.
+#' @return
+#' A dataset with one additional column.
+#' @export
+#' @importFrom rlang :=
+#' @examples
+#' test_data <- data.frame(
+#' rCSILessQlty = c(1,2,3,1),
+#' rCSIBorrow = c(0,0,3,0),
+#' rCSIMealSize = c(4,2,6,1),
+#' rCSIMealAdult = c(4,3,5,0),
+#' rCSIMealNb = c(2,5,NA_integer_,1)
+#' )
+#' add_rcsi(test_data)
+
+
+
+add_rcsi <- function(data,
+                     rCSILessQlty = "rCSILessQlty",
+                     rCSIBorrow = "rCSIBorrow" ,
+                     rCSIMealSize = "rCSIMealSize",
+                     rCSIMealAdult = "rCSIMealAdult",
+                     rCSIMealNb = "rCSIMealNb",
+                     new_colname = "rcsi"
+){
+
+  score_name <- paste0(new_colname, "_score")
+  cat_name <- paste0(new_colname, "_cat")
```

Comment:
I think the flexibility in naming is a good characteristic, but also please check in on the add_fcm_phase function to ensure interoperability. I suggest either harmonize to the colname rcsi_cat so that function can smoothly use this output, or alternatively suggest in add_fcm_phase to have additional flexibility in the column name for rcsi_cat that is given as an input... because right now there is a risk I think that add_fcm_phase cannot use the output of this function, at least not smoothly.

Reply:
Now that Abraham is changing the function, i think it should be fine. Also, I am using the same default value that Abraham using, so there should be no issue, as it will create `rcsi_cat` unless the user changes the name from their side. But I would like the keep the option open for the user so they can change the name if necessary.